test/e2e/network/ingress.go

```diff
@@ -1114,17 +1114,23 @@ var _ = SIGDescribe("Ingress API", func() {
 		ginkgo.By("deleting")
 		err = ingClient.Delete(context.TODO(), createdIngress.Name, metav1.DeleteOptions{})
 		framework.ExpectNoError(err)
-		_, err = ingClient.Get(context.TODO(), createdIngress.Name, metav1.GetOptions{})
-		framework.ExpectEqual(apierrors.IsNotFound(err), true, fmt.Sprintf("expected 404, got %#v", err))
+		ing, err := ingClient.Get(context.TODO(), createdIngress.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNotEqual(ing.DeletionTimestamp, nil, fmt.Sprint("expected deletionTimestamp, got nil"))
 		ings, err = ingClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(len(ings.Items), 2, "filtered list should have 2 items")
+		// Should have 3 items since createdIngress will not have been deleted yet due to the finalizer
+		framework.ExpectEqual(len(ings.Items), 3, "filtered list should have 3 items")
 
 		ginkgo.By("deleting a collection")
 		err = ingClient.DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
 		ings, err = ingClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(len(ings.Items), 0, "filtered list should have 0 items")
+		for _, ing := range ings.Items {
+			framework.ExpectNotEqual(ing.DeletionTimestamp, nil, fmt.Sprint("expected deletionTimestamp, got nil"))
+		}
+		// Should have 3 items since the ingresses will not have been deleted yet due to the finalizer
+		framework.ExpectEqual(len(ings.Items), 3, "filtered list should have 3 items")
```

Comment:
we can't assume there will be a finalizer. we can assert the list length is <= 3, and that any existing items should have a deletionTimestamp and a finalizer

Reply:
done